zigen-project/zen
# Configure bg images with .config/zen-desktop/config.toml

## Context
To realize user-specific configuration like #123, we need a way to give configuration data to zen-desktop after the build.

## Summary
Let users specify their background image with a `toml` config file.Following [XDG Base Directory Specification
](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), it first searches `$XDG_CONFIG_HOME/zen-desktop/config.toml`, then falls back to `$HOME/.config/zen-desktop/config.toml`.

## How to check behavior
- Create the following config file and save it as `$HOME/.config/zen-desktop/config.toml`. Note that only **PNG images with RGBA format** can be used as the background images for now.
```toml
[background]
image = "/path/to/your/image/file"
```
- Run `zen-desktop` and see if the change is reflected.
